File: TheReader/Models/RecentSearch.swift
import Foundation
import SwiftData

@Model
final class RecentSearch {
    @Attribute(.unique) var id: String
    var query: String
    var source: MangaSource
    var timestamp: Date

    init(
        id: String = UUID().uuidString,
        query: String,
        source: MangaSource = .mangadex,
        timestamp: Date = .now
    ) {
        self.id = id
        self.query = query
        self.source = source
        self.timestamp = timestamp
    }
}

File: flutter_app/lib/models/recent_search.dart
import 'package:hive/hive.dart';
import 'manga.dart';

part 'recent_search.g.dart';

@HiveType(typeId: 5)
class RecentSearch extends HiveObject {
  @HiveField(0)
  final String id;

  @HiveField(1)
  final String query;

  @HiveField(2)
  final MangaSource source;

  @HiveField(3)
  DateTime timestamp;

  RecentSearch({
    required this.id,
    required this.query,
    this.source = MangaSource.mangadex,
    DateTime? timestamp,
  }) : timestamp = timestamp ?? DateTime.now();
}
